dts/arm/nordic/nrf52840.dtsi

```diff
@@ -55,6 +55,32 @@
 			label = "UART_1";
 		};
 
+		gpiote0: gpiote@40006000 {
+			compatible = "nordic,nrf5-gpiote";
+			reg = <0x40006000 0x1000>;
+			interrupts = <6 5>;
+			status = "disabled";
+			label = "GPIOTE_0";
+		};
+
+		gpio0: gpio@50000000 {
+			compatible = "nordic,nrf5-gpio";
+			gpio-controller;
+			reg = <0x50000000 0x1000>;
```

Comment:
The gpio0 memory region (reg property) overlaps with the gpio1 memory region.

Reply:
Hi, thanks for reviewing.

Like I said earlier, the regions for the two must overlap if we want the reg address to be the base address of the IP block, because the base address of gpio1 is in the gpio0 range:

0x5000_0000 --> gpio0 base address
0x5000_0300 --> gpio1 base address
0x5000_0504 --> first gpio0 register
0x5000_0804 --> first gpio1 register

This is why in the first go around I had the addresses as 0x50000504 and 0x50000804, although this doesn't make much sense when looking at it from the generated defines/driver perspective.

Options:
- ignore the problem
- make the size of the reg 0, (or leave the size out?)
- go back to the way I first had it
- make gpio0 and gpio1 children of a new node type:
```
gpio {
    compatible="nordic,nrf5-gpio-????";
    label = "????";
    reg = <0x50000000 0x1000>;

    gpio0 {
       compatible="nordic-nrf5-gpio";
       offset = <0x504>;  
       gpio-controller;
       label = "GPIO0";
    };
    gpio1 {
       compatible="nordic-nrf5-gpio";
       offset = <0x804>;
       gpio-controller;
       label = "GPIO1"
    };
};
```

Reply:
@marc-cpdesign: Thank you for clear explanation. Now I see the problem.  In such situation we should declare several regions in the reg property:

```
gpio0: gpio@50000000 {
        reg = <0x50000000 0x300
                    0x50000500 0x300>;
}
```

However I do not know if such declarations are supported by the Zephyr DTS infrastructure.
@nvlsianpu, @carlescuf: Do you know if our infrastructure supports entries like the one above?

Reply:
This is what I've got:
```
/* gpio@50000000 */
#define NORDIC_NRF5_GPIO_50000000_BASE_ADDRESS_0	0x50000000
#define NORDIC_NRF5_GPIO_50000000_BASE_ADDRESS_1	0x50000500
#define NORDIC_NRF5_GPIO_50000000_LABEL			"GPIO_0"
#define NORDIC_NRF5_GPIO_50000000_SIZE_0		768
#define NORDIC_NRF5_GPIO_50000000_SIZE_1		768
```
@pizi-nordic  so It is supported.

Reply:
@nvlsianpu Thanks for that. I will make changes shortly.